### DISCUSSION TITLE
homeSpan.setWifiCallback   stops accessory being added to Home

### DISCUSSION THREAD

**Discussion by TomInIowa** [HomeSpan/HomeSpan#287] [Q&A] [answered]
I have this arduino sketch that works for 4  ESP32 boards, but on the **5th one it does not**.
If I comment out two lines, then the device can be added as an accessory.  
               // homeSpan.setPortNum(1201);           
              // homeSpan.setWifiCallback(setupWeb);    
Any suggestions on what/where  to look for?
-------------
...what I have tried
serial monitor "U" to make  the board not paired
             *** HomeSpan Pairing Data DELETED ***
In Home, when I add accessory and select the one I want to work , I get 
            ** Client #0 Connected: (108 sec) 192.168.2.13 on Socket 4/16
            In Pair Setup...    ip address: 192.168.2.170
in the serial monitor. 
If I don't comment out the two lines, I do not see 
             "** Client #0 Connected: (108 sec) 192.168.2.13 on Socket 4/16
I             n Pair Setup...    ip address: 192.168.2.170
"
My problem seems to have to do with the command for the webpage.  Yet it does work for 4 other boards. Everything is the same except for the hostname and accessory name from one board to another.

**Comment by HomeSpan** [comment]
Are all the boards identical?  Which sketch are you using - the Programmable Hub Example?  Does the 5th board work if you leave in setPortNum but take out the setWifiCallback?  One thing to try is the 'H' command from the Serial Monitor.  This resets the Device ID on the board which ensures the Home App will not try to use any previously cached info.

**Reply by TomInIowa** [reply]
I've had a set back.  I upgraded my mac os yesterday to 12.3 
It killed off the Arduino ide.
exec: "python": executable file not found in $PATH
Error compiling for board Adafruit ESP32 Feather.

If I change the board to  Generic  ESP8266, then it works.

**Reply by HomeSpan** [reply]
This seems to be a general issue with the Arduino-ESP32 library and changes from 12.2 to 12.3 unrelated to HomeSpan.  I have not yet upgraded to 12.3 but will do so in the next week or so.   There are number of sites reporting this issue with various workarounds - problem stems from python 2.X being dropped from MacOS (only python 3.X is included).

**Reply by Xorlent** [reply]
I addressed the issue by setting my PATH to include Python 3.9 (lots of online articles on how to install Python on MasOS - homebrew is the simplest).  Once Python is installed, just edit ~/.zshenv and add a directive that points to the location of the Python executable:

In my case, I added this line to .zshenv:
export PATH=$PATH:/opt/homebrew/bin

Also, when you use homebrew to install Python, you may need to also create a symbolic link to Python 3:
sudo ln -s /opt/homebrew/bin/python3 /opt/homebrew/bin/python

Now, until the issue is fixed, you have to start Arduino IDE from a terminal, but this does fix the problem so you can move on.

To launch (this assumes you have it installed in the user's Application folder):
open ~/Applications/Arduino.app

To launch (machine-wide install):
open /Applications/Arduino.app

**Comment by TomInIowa** [comment, marked as answer]
I backed up (in versions numbers ) to a sketch that did work.  Then I kept added parts of the new code to my sketch.  I got all the way back to my lastest sketch and everything is still working.   Maybe it was a Apple HomeKit  problem... it had some left over pairing information?       Thank you for the "Programmable Hub Example".  That is where I started.
